python/activator/activator.py

```diff
@@ -195,6 +227,113 @@ def create_app():
+                          groupId=obj['groupId'],
+                          coordinateSystem=obj['coordinateSystem'],
+                          position=obj['position'],
+                          startTime=obj['startTime'],
+                          rotationSystem=obj['rotationSystem'],
+                          cameraAngle=obj['cameraAngle'],
+                          filters=obj['filters'],
+                          dome=obj['dome'],
+                          duration=obj['duration'],
+                          nimages=obj['nimages'],
+                          instrument=obj['instrument'],
+                          survey=obj['survey'],
+                          totalCheckpoints=obj['totalCheckpoints'],
+                          detector=obj['detector'],
+                          private_sndStamp=obj['private_sndStamp'])
+
+
+def keda_start():
+
+    setup_usdf_logger(
+        labels={"instrument": instrument_name},
+    )
+
+    # Initialize local registry
+    registry = activator.repo_tracker.LocalRepoTracker.get()
+    registry.init_tracker()
+
+    # Check initialization and abort early
+    _get_consumer()
+    _get_storage_client()
+    _get_central_butler()
+    _get_local_repo()
+
+    # Initiialize schema registry for fan out
+    fan_out_schema_registry_conf = {'url': fan_out_schema_registry_url}
+    fan_out_schema_registry_client = SchemaRegistryClient(fan_out_schema_registry_conf)
+
+    fan_out_avro_deserializer = AvroDeserializer(schema_registry_client=fan_out_schema_registry_client,
+                                                 from_dict=dict_to_fanned_out_visit)
+
+    fan_out_consumer_conf = {
+        "bootstrap.servers": fan_out_kafka_cluster,
+        "group.id": fan_out_kafka_group_id,
+        "auto.offset.reset": fan_out_kafka_topic_offset,
+        "sasl.mechanism": fan_out_kafka_sasl_mechanism,
+        "security.protocol": fan_out_kafka_security_protocol,
+        "sasl.username": fan_out_kafka_sasl_username,
+        "sasl.password": fan_out_kafka_sasl_password,
+        'enable.auto.commit': False
+    }
+
+    fan_out_consumer = kafka.Consumer(fan_out_consumer_conf, logger=_log)
+
+    _log.info("starting fan out consumer")
+
+    try:
+        fan_out_consumer.subscribe([fan_out_kafka_topic])
+
+        fan_out_listen_start_time = time.time()
+
+        while time.time() - fan_out_listen_start_time < fanned_out_msg_listen_timeout:
+            fan_out_consumer.subscribe([fan_out_kafka_topic])
+            fan_out_message = fan_out_consumer.poll(timeout=5)
+            if fan_out_message is None:
+                continue
+            if fan_out_message.error():
+                if fan_out_message.error().code() == KafkaError._PARTITION_EOF:
+                    # End of partition event
+                    _log.warning('reached end of offset')
+                elif fan_out_message.error():
+                    raise KafkaException(fan_out_message.error())
```

Comment:
Is is really a good idea to abort on any Kafka error? We get lots of harmless errors while listening for raw notifications, that's what the `_filter_messages` function is for.